packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -533,57 +483,42 @@ export class NftDetectionController extends StaticIntervalPollingController<
   #onPreferencesControllerStateChange({ useNftDetection }: PreferencesState) {
     if (!useNftDetection !== this.#disabled) {
       this.#disabled = !useNftDetection;
-      if (useNftDetection) {
-        // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-        this.start();
-      } else {
-        this.stop();
-      }
     }
   }
 
-  #getOwnerNftApi({ address, next }: { address: string; next?: string }) {
-    // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+  #getOwnerNftApi({
+    chainId,
+    address,
+    next,
+  }: {
+    chainId: string;
+    address: string;
+    next?: string;
+  }) {
     // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```

Comment:
Looks like this can be removed.

```suggestion
```

Reply:
I was able to remove this by adding a type assertion `as string` to the constant.
I thought the type string should be trivially inferred from the constant but it seems it is not the case, that might have to do with our linting rules.

Reply:
Hmm I'm seeing different results for this on local. I'm good with going with this since it's passing CI.